File: src/main/java/com/max/student/Student.java
package com.max.student;

public class Student {
    String id; //initial value =null
    String name;
    int english; //initial value=0;
    int math;
    static int pass = 60;
    public Student(){
        super();
    }

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest(){

//        int max = (english > math) ? english : math;
//        if (english > math){
//            max=english;
//        } else {
//            max=math;
//        }
       return (english > math) ? english : math;
//        return max;
    }

    public void print(){
        int average = getAverage();
        System.out.print(name + "\t" + english +"\t" + math +
                "\t" + getAverage()+ "\t" + ((getAverage()>=pass) ? "PASS" : "Failed"));
        char grading = 'F';
        switch (average/10){ //小數點後會去除
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
        }
        /*if (average >=90 && average <=100) {
            grading = 'A';
        }else if (average>=80 && average<=89){
            grading= 'B';
        }else if (average>=70 && average<=79) {
            grading = 'C';
        }else if (average>=60 && average<=69) {
            grading = 'D';
        }*/
        System.out.println("\tgrading is:" + grading);
        /*if (getAverage()>=60){
            System.out.println("\tPASS");
        }else{
            System.out.println("\tFailed");
        }*/
    }
//    public Student(String name  , int english, int math) {
//        this.name = name;//把區域變數放到屬性身上
//        this.english = english;
//        this.math = math;
//    }
    public int getAverage(){
        return (english+math)/2;
    }
}

File: src/main/kotlin/com/max/student/Student.kt
package com.max.student

import java.util.*

fun main(args: Array<String>) {
//    userInout()
    val stu = Student("Jack_KT", 60, 99)
    stu.print();
    val test =123
    println("There is : $test")
    println("Highest score: ${stu.highest()}" )
    println("Highest score: " + stu.highest())
    }

class Student(var name:String?, var english:Int, var math:Int){
    fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
        /*print(name +"\t"+english +"\t" +math+"\t"+getAverage() + "\t" + passOrFailed())
        println("\t" + grading())*/
    }
    fun passOrFailed() = if(getAverage()>=60) "PASS" else "failed"
    fun grading() = when(getAverage()) {
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'D'
            else -> 'F'
          }
    fun getAverage() = (english+math)/2;

    fun highest()  = if (english > math) {
            println("english")
            english
        } else {
            println("math")
            math
        };
//    if (english > math) {
//        max = english;
//    } else {
//        max = math;
//    }

    fun nameCheck()= name?.length

}



private fun userInput() {
    val scanner = Scanner(System.`in`)
    println("Enter name:")
    var name = null
//    var name = scanner.next()
    print("Enter englih's score=")
    var english = scanner.nextInt()
    print("Enter math's score=")
    var math = scanner.nextInt()

    val stu = Student(name, english, math)
    stu.print();
    stu.nameCheck();
}

